contracts/Datastore.sol

```diff
@@ -134,16 +139,48 @@ contract Datastore is AragonApp {
         writeAccess = hasWriteAccess(_fileId, _caller);
     }
 
-
+    /**
+     * @notice Returns the encryption key for file with `_fileId`
+     * @param _fileId File Id 
+     */
+    function getFileEncryptionKey(uint _fileId) external view returns(string) {
+        if (hasReadAccess(_fileId, msg.sender)) {
+            FileLibrary.File storage file = fileList.files[_fileId];
+            return file.cryptoKey;
+        }
+        return "0";
+    } 
 
     /**
-     * @notice Delete file with Id `_fileId`
+     * @notice Set file `_fileId` as deleted or not.
      * @param _fileId File Id
+     * @param _isDeleted Is file deleted or not
+     * @param _deletePermanently If true, will delete file permanently
      */
-    function deleteFile(uint _fileId) public onlyFileOwner(_fileId) {
-        fileList.deleteFile(_fileId);
+    function deleteFile(uint _fileId, bool _isDeleted, bool _deletePermanently) public onlyFileOwner(_fileId) {
+        if (_isDeleted && _deletePermanently) {
+            fileList.permanentlyDeleteFile(_fileId);
+            emit DeleteFilePermanently(msg.sender);            
+        }
+        else {
+            fileList.setIsDeleted(_fileId, _isDeleted);
+            emit DeleteFile(msg.sender);
+        }
     }
 
+    /**
+     * @notice Delete files in `_fileIds`. Files cannot be restored
+     * @param _fileIds File Ids
+     */
+    function deleteFilesPermanently(uint256[] _fileIds) public {
+        for(uint256 i = 0; i < _fileIds.length; i++)
+            fileList.permanentlyDeleteFile(_fileIds[i]);
+        emit DeleteFilePermanently(msg.sender);
```

Comment:
In case this event finally includes `fileId` (see previous comment), then it should happen at per file level.

Reply:
See new commits cherry picked from #86